docs/SerhiiNSN/scripts/games.js

```diff
@@ -0,0 +1,103 @@
+const template = document.querySelector('[data-type="card-template"]');
+let allGamesData = [];
+let gamesDisplayed = [];
+const checkboxNew = document.querySelector('[data-filter="new"]');
+const checkboxOld = document.querySelector('[data-filter="old"]');
+const loadingOverlay = document.querySelector('.loading-overlay');
+
+function showLoadingOverlay() {
+    loadingOverlay.style.display = 'block';
+}
+
+function hideLoadingOverlay() {
+    loadingOverlay.style.display = 'none';
+}
+
+async function createCardElement(game) {
+    const cardCopy = document.createElement('li');
+    cardCopy.classList.add('games__card');
+    cardCopy.innerHTML = template.innerHTML;
+
+    cardCopy.querySelector('[data-card-img]').src = game.thumbnail;
+    cardCopy.querySelector('[data-card-title]').textContent = game.title;
+    cardCopy.querySelector('[data-card-release]').innerHTML = `<span class='bold'>Release_date:</span> ${game.release_date}`;
+
+    return cardCopy;
+}
+
+async function renderCards(container, games) {
+    const fragment = document.createDocumentFragment();
+    const containerElement = document.querySelector(`[data-type="${container}"]`);
+
+    await Promise.all(games.map(async (game) => {
```

Comment:
... then you can skip this `await Promise.all` and make the current function synchronous too.